makefiles/examples/Makefile

```diff
@@ -0,0 +1,58 @@
+.PHONY: setup
+setup: ## Install all the build and lint dependencies
+	go get -u github.com/alecthomas/gometalinter
+	go get -u golang.org/x/tools/cmd/cover
+	go get -u github.com/golang/dep/cmd/dep
+	gometalinter --install --update
+	@$(MAKE) dep
+
+.PHONY: dep
+dep: ## Run dep ensure and prune
+	dep ensure
+	dep prune
+
+.PHONY: test
+test: ## Run all the tests
+	echo 'mode: atomic' > coverage.txt && go list ./... | grep -v /vendor/ | xargs -n1 -I{} sh -c 'go test -covermode=atomic -coverprofile=coverage.txt -v -race -timeout=30s {}'
+
+.PHONY: cover
+cover: test ## Run all the tests and opens the coverage report
+	go tool cover -html=coverage.txt
+
+.PHONY: fmt
+fmt: ## gofmt and goimports all go files
+	find . -name '*.go' -not -wholename './vendor/*' | while read -r file; do gofmt -w -s "$$file"; goimports -w "$$file"; done
```

Comment:
Is this now solved thanks to the updates which mean `./...` now exclude the vendor dir? Or do we need to upgrade to go1.9 first?

Reply:
will test

Reply:
still doesnt work :(

Althought since `goimports` also `gofmts`, we can remove the call to `gofmt`